File: eshop/src/main/java/eshop/entities/Adresse.java
package eshop.entities;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

import eshop.entities.jsonviews.JsonViews;

@Embeddable
public class Adresse {
	@JsonView(JsonViews.Simple.class)
	private String numero;
	@Column(name = "street")
	@JsonView(JsonViews.Simple.class)
	private String rue;
	@JsonView(JsonViews.Simple.class)
	private String codePostal;
	@JsonView(JsonViews.Simple.class)
	private String ville;

	public Adresse() {

	}

	public Adresse(String numero, String rue, String codePostal, String ville) {
		super();
		this.numero = numero;
		this.rue = rue;
		this.codePostal = codePostal;
		this.ville = ville;
	}

	public String getNumero() {
		return numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}

	public String getRue() {
		return rue;
	}

	public void setRue(String rue) {
		this.rue = rue;
	}

	public String getCodePostal() {
		return codePostal;
	}

	public void setCodePostal(String codePostal) {
		this.codePostal = codePostal;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	@Override
	public int hashCode() {
		return Objects.hash(codePostal, numero, rue, ville);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Adresse other = (Adresse) obj;
		return Objects.equals(codePostal, other.codePostal) && Objects.equals(numero, other.numero)
				&& Objects.equals(rue, other.rue) && Objects.equals(ville, other.ville);
	}

}

File: eshopAngular/src/app/model/adresse.ts
export class Adresse {
  public get ville(): string | undefined {
    return this._ville;
  }
  public set ville(value: string | undefined) {
    this._ville = value;
  }
  public get codePostal(): string | undefined {
    return this._codePostal;
  }
  public set codePostal(value: string | undefined) {
    this._codePostal = value;
  }
  public get rue(): string | undefined {
    return this._rue;
  }
  public set rue(value: string | undefined) {
    this._rue = value;
  }
  public get numero(): string | undefined {
    return this._numero;
  }
  public set numero(value: string | undefined) {
    this._numero = value;
  }
  public constructor(
    private _numero?: string,
    private _rue?: string,
    private _codePostal?: string,
    private _ville?: string
  ) {}
}
